### ISSUE TITLE
** (EXIT) %Tds.Error{message: "tcp closed", mssql: nil}

### ISSUE THREAD

**Issue by Sharespot** [livehelpnow/tds_ecto#38]
Hi, 

I have a virtual machine in Azure with Phoenix Frmaework and I am trying to connect with my SQL Database (Azure relational database-as-a-service). I follow all the steps to set my application with MSSQL database, but when I try to "MIX_ENV=prod mix ecto.migrate" I receive this error: 

21:40:16.348 [error] GenServer #PID<0.191.0> terminating
*\* (stop) %Tds.Error{message: "tcp closed", mssql: nil}
*\* (exit) exited in: GenServer.call(#PID<0.191.0>, {:query, "IF NOT EXISTS (SELECT \* FROM information_schema.tables t WHERE t.table_name = 'schema_migrations') BEGIN CREATE TABLE [schema_migrations]([version] bigint NOT NULL PRIMARY KEY, [inserted_at] datetime2 NULL)END", [], [timeout: :infinity]}, :infinity)
    *\* (EXIT) %Tds.Error{message: "tcp closed", mssql: nil}
    (elixir) lib/gen_server.ex:604: GenServer.call/3
    (tds) lib/tds/connection.ex:62: Tds.Connection.call_proc/3
    lib/tds_ecto/connection.ex:59: Tds.Ecto.Connection.query/4
    (ecto) lib/ecto/adapters/sql.ex:259: Ecto.Adapters.SQL.query/7
    (ecto) lib/ecto/pool.ex:159: Ecto.Pool.do_run/4
    (ecto) lib/ecto/adapters/sql.ex:247: Ecto.Adapters.SQL.query/6
    (ecto) lib/ecto/adapters/sql.ex:225: Ecto.Adapters.SQL.query/5
    lib/tds_ecto.ex:104: Tds.Ecto.execute_ddl/3

My prod.secret.exs database settings is like that:
# Configure your database

config :phoenix_sample, PhoenixSample.Repo,
  adapter: Tds.Ecto,
  username: "username",
  password: "password",
  database: "StagingRefactory_db",
  hostname: "stagingrefactory-dbserver.database.windows.net",
  pool_size: 20

Could anyone help me?

**Comment by brandon8088** [comment]
I have gotten that error when db permissions were incorrect hitting a standard mssql db (not managed azure db).

**Comment by andrei-mihaila** [comment]
I got the same issue with the tds package (the package that tds_ecto uses). After playing around with the code I found out that the server did respond with an error message (and then closed the connection) but unfortunately the code doesn't catch it (this needs to be fixed, I will try my hand at it when I can).

In my case the error that the server returned was:
> Server name cannot be determined.  It must appear as the first segment of the server’s dns name (servername.mscds.com).  Some libraries do not send the server name, in which case the server name must be included as part of the user name (username@servername).  In addition, if both formats are used, the server names must match.

As suggested in [one of their troubleshooting guides](https://blogs.msdn.microsoft.com/sqlnativeclient/2010/11/30/troubleshooting-sql-azure-applications-with-sql-server-native-client/) I changed the username to the `user@server` format. In your case this should be `username@stagingrefactory-dbserver.database.windows.net`. Hope it fixes the problem for you!
